src/message.py

```diff
@@ -24,36 +25,53 @@ def __init__(self, code, body=None):
 
     def send_response(self):
         """Format reponse for in http protocol."""
-        message = '{0} {1}{2}{3}'.format(
+        # message = '{0} {1}{2}{3}'.format(
+        #     HTML_PROTOCOL,
+        #     RESPONSE_CODE[self.code],
+        #     CRLF + CRLF,
+        #     self.body)
+        message = '{0} {1}{2}'.format(
             HTML_PROTOCOL,
             RESPONSE_CODE[self.code],
-            CRLF + CRLF,
-            self.body)
-        return message.encode('utf8')
+            CRLF + CRLF)
+        message = message.encode('utf-8')
+        try:
+            message += self.body[1]
+        except TypeError:
+            if self.code == 200:
```

Comment:
It looks like you're relying on the type of data in self.body[1] in order to determine if you're supposed to be returning a directory listing.  This feels janky to me.  A directory listing should be a bytestring just like any other response body.